src/Microsoft.OData.Client/ALinq/QueryableResourceExpression.cs

```diff
@@ -507,9 +507,9 @@ internal void SetKeyPredicate(IEnumerable<Expression> keyValues)
         /// Gets the key properties from KeyPredicateConjuncts
         /// </summary>
         /// <returns>The key properties.</returns>
-        internal Dictionary<PropertyInfo, ConstantExpression> GetKeyProperties()
+        internal IList<KeyValuePair<PropertyInfo, ConstantExpression>> GetKeyProperties()

```

Comment:
Check if the returned collection is modified elsewhere. If not, the return type can be `IReadOnlyList<KeyValuePair<PropertyInfo, ConstantExpression>>`

Reply:
If I change it, it seems I have to change other methods' syntax. Since it's internal and not a big deal, let me keep it unchanged.